### PR TITLE
CI: Support more version formats in publishing

### PR DESCRIPTION
**Pull request by kminehart**
* Removed unused commands from `pkg/build`
* Removed dead code from `pkg/build`
* Updated the grafana.com publish step to allow more version formats
* Centralize the file name generation into 1 universal function
* Add lots of tests

### CODE REVIEW

File: pkg/build/cmd/grafanacom.go [hunk 1]
```diff
@@ -186,15 +226,26 @@ func publishPackages(cfg packaging.PublishConfig) error {
 		r.ReleaseNotesURL = cfg.ReleaseNotesURL
 	}
 
+	// 1. Create the release
 	if err := postRequest(cfg, "versions", r, fmt.Sprintf("create release %s", r.Version)); err != nil {
 		return err
 	}
 
+	// 2.
```
Comment (baldm0mma) [inline pkg/build/cmd/grafanacom.go:234]: Is there a comment you have for this?

**Review by zserge — APPROVED** [review]

File: pkg/build/cmd/grafanacom.go (hunk 1 above)
Comment (kminehart) [reply pkg/build/cmd/grafanacom.go:234]: is this not descriptive enough!?

Comment (kminehart) [reply pkg/build/cmd/grafanacom.go]: (joke, removed that comment)

**Review by baldm0mma — APPROVED** [review]
🏋️‍♂️🏋️‍♂️🏋️‍♂️

**Pull request merged**